Tweaked alsaconf to work properly with 2.6 pci drivers.
Elimar Riesebieter 2007/09/11


Index: alsa-utils-1.0.18/alsaconf/alsaconf.in
===================================================================
--- alsa-utils-1.0.18.orig/alsaconf/alsaconf.in	2008-11-02 20:59:26.000000000 +0100
+++ alsa-utils-1.0.18/alsaconf/alsaconf.in	2008-11-02 20:59:26.000000000 +0100
@@ -415,9 +415,9 @@
 set Front -12dB
 set PCM 90% unmute
 set PCM 0dB
-mixer Synth 90% unmute
+mixer Synth 75% unmute
 mixer Synth 0dB
-mixer CD 90% unmute
+mixer CD 75% unmute
 mixer CD 0dB
 # mute mic
 set Mic 0% mute
@@ -425,22 +425,22 @@
 set PCM,1 90% unmute
 set PCM,1 0dB
 # Trident/YMFPCI/emu10k1
-set Wave 100% unmute
-set Music 100% unmute
-set AC97 100% unmute
+set Wave 75% unmute
+set Music 75% unmute
+set AC97 50% unmute
 # CS4237B chipset:
 set 'Master Digital' 75% unmute
 # Envy24 chips with analog outs
-set DAC 90% unmute
+set DAC 75% unmute
 set DAC -12dB
-set DAC,0 90% unmute
+set DAC,0 75% unmute
 set DAC,0 -12dB
-set DAC,1 90% unmute
+set DAC,1 75% unmute
 set DAC,1 -12dB
 # some notebooks use headphone instead of master
 set Headphone 75% unmute
 set Headphone -12dB
-set Playback 100% unmute
+set Playback 75% unmute
 # turn off digital switches
 set "SB Live Analog/Digital Output Jack" off
 set "Audigy Analog/Digital Output Jack" off
@@ -458,8 +458,16 @@
     Advanced Linux Sound Architecture (ALSA) driver.
 
 
-  You should stop all sound applications now." $version)
-  $DIALOG --msgbox "$msg" 20 63 || acex 0
+  You should stop all sound applications now.
+
+  Please notice, that sound configuration with this script
+  is not needed if you are running testing/sid with an up
+  to date kernel and your sound device is a suopported PCI
+  one. In that case *udev* is predestinated to load your
+  driver ;)
+  But any way, you can finish this procedure with a working
+  PCI-Sound at the end." $version)
+  $DIALOG --msgbox "$msg" 24 63 || acex 0
 }
 
 # FAREWELL
@@ -478,7 +486,7 @@
      program such as alsamixer or gamix.
   
   ")
-  $DIALOG --msgbox "$msg" 17 60 || acex 0
+  $DIALOG --msgbox "$msg" 19 60 || acex 0
 }
 
 clear() {
@@ -983,14 +991,15 @@
     if [ "$distribution" = "gentoo" ]; then
       xecho "Running modules-update..."
       modules-update
+      if [ -x $rcalsasound ] ; then
+        echo Loading driver...
+        $rcalsasound restart
+      fi
     elif [ "$distribution" = "debian" ]; then
-      xecho "Running update-modules..."
-      update-modules
-    fi
-    if [ -x $rcalsasound ] ; then
-      echo Loading driver...
-      $rcalsasound restart
+      xecho "Loading driver..."
+      /sbin/modprobe $CARD_DRIVER
     fi
+
     echo Setting default volumes...
     if [ -x $bindir/set_default_volume ]; then
 	$bindir/set_default_volume -f
